src/rgw/driver/rados/rgw_sal_rados.cc

```diff
@@ -1109,6 +1110,61 @@ int RadosStore::remove_topics(const std::string& tenant, RGWObjVersionTracker* o
       objv_tracker, y);
 }
 
+int RadosStore::read_topic(const std::string& topic_name,
+                           rgw_pubsub_topic& topic,
+                           RGWObjVersionTracker* objv_tracker,
+                           optional_yield y,
+                           const DoutPrefixProvider* dpp) {
```

Comment:
v1 topics are isolated by `tenant` namespaces - see `topics_oid()` which uses a separate rados object for each tenant name. i think v2 topics need similar namespace isolation, so reads should require both `topic_name` and `tenant`

Reply:
> v1 topics are isolated by `tenant` namespaces - see `topics_oid()` which uses a separate rados object for each tenant name. i think v2 topics need similar namespace isolation, so reads should require both `topic_name` and `tenant`

done